### PR TITLE
API: replace BorutaPy with ARFS

### PR DESCRIPTION
**Pull request by j-ittner**
BorutaPy is no longer maintained and is not compatible with current versions of Numpy.

Adding support for ARFS as a replacement for BorutaPy.

### CODE REVIEW

File: src/sklearndf/wrapper/_wrapper.py [hunk 1]
```diff
@@ -1211,6 +1220,13 @@ class MetaEstimatorWrapperDF(
     copy of the `X` and `y` parameters, so we need to make sure that these are converted
     back to data frames.
 
+    We also ensure that the embedded estimator is a native sci-kit learn estimator.
```
Comment (mtsokol) [inline src/sklearndf/wrapper/_wrapper.py:1234]: Just started reviewing the PR - only a nitpick here: `sci-kit` to `scikit-learn`

Comment (j-ittner) [reply src/sklearndf/wrapper/_wrapper.py:1234]: Good catch, thanks @mtsokol !

File: src/sklearndf/wrapper/_wrapper.py [hunk 2]
```diff
@@ -616,7 +617,15 @@ def __setattr__(self, name: str, value: Any) -> None:
         if name.startswith("_"):
             super().__setattr__(name, value)
```
Comment (mtsokol) [inline src/sklearndf/wrapper/_wrapper.py:625]: Shouldn't we raise an exception here? I see that the comment two lines above indicates that only public attributes can be set here.
Right now it's the same as in line `628`. 
Also `__getattr__` raises an exception for accessing private attributes (line `603`).

**Review by mtsokol — APPROVED** [review]
Looks good to me! Only one question left.

File: src/sklearndf/wrapper/_wrapper.py (hunk 2 above)
Comment (j-ittner) [reply src/sklearndf/wrapper/_wrapper.py:625]: Thanks for keeping a close eye on this critical piece of the wrapper code. I have updated the inline comments, hope that makes it clearer. I think the code is ok as-is.

Comment (mtsokol) [reply src/sklearndf/wrapper/_wrapper.py:625]: Ah, now it's clear! Thanks for explanation!

**Pull request merged**